pkg/ibi/imageclusterinstall.go

```diff
@@ -0,0 +1,497 @@
+	if valid, _ := builder.validate(); !valid {
+		return nil
+	}
+
+	if clusterDeploymentName == "" {
+		glog.V(100).Infof("The imageclusterinstall clusterdeployment is empty")
+
+		builder.errorMsg = "imageclusterinstall clusterdeployment cannot be empty"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.ClusterDeploymentRef == nil {
+		builder.Definition.Spec.ClusterDeploymentRef = &corev1.LocalObjectReference{}
+	}
+
+	builder.Definition.Spec.ClusterDeploymentRef.Name = clusterDeploymentName
+
+	return builder
+}
+
+// WithExtraManifests includes manifests via configmap name.
+func (builder *ImageClusterInstallBuilder) WithExtraManifests(extraManifestsName string) *ImageClusterInstallBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return nil
+	}
+
+	if extraManifestsName == "" {
+		glog.V(100).Infof("The imageclusterinstall extramanifest is empty")
+
+		builder.errorMsg = "imageclusterinstall extramanifest cannot be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.ExtraManifestsRefs =
+		append(builder.Definition.Spec.ExtraManifestsRefs, corev1.LocalObjectReference{
+			Name: extraManifestsName,
+		})
+
+	return builder
+}
+
+// WithMachineNetwork specifies the machine network where nodes will be installed.
+func (builder *ImageClusterInstallBuilder) WithMachineNetwork(network string) *ImageClusterInstallBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return nil
+	}
+
+	if _, _, err := net.ParseCIDR(network); err != nil {
+		glog.V(100).Infof("The machinenetwork is not a properly formatted IP network address")
+
+		builder.errorMsg = "imageclusterinstall machinenetwork incorrectly formatted"
+
+		return builder
+	}
+
+	builder.Definition.Spec.MachineNetwork = network
+
+	return builder
+}
+
+// WithSSHKey adds specified ssh key to authorized_keys of installed nodes.
+func (builder *ImageClusterInstallBuilder) WithSSHKey(sshKey string) *ImageClusterInstallBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return nil
+	}
+
+	if sshKey == "" {
+		glog.V(100).Infof("The imageclusterinstall sshkey is empty")
+
+		builder.errorMsg = "imageclusterinstall sshkey cannot be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.SSHKey = sshKey
+
+	return builder
+}
+
+// WithVersion sets the seedimage version for the imageclusterinstall.
+func (builder *ImageClusterInstallBuilder) WithVersion(version string) *ImageClusterInstallBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return nil
+	}
+
+	if version == "" {
+		glog.V(100).Infof("The imageclusterinstall version is empty")
+
+		builder.errorMsg = "imageclusterinstall version cannot be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.Version = version
+
+	return builder
+}
+
+// GetCompletedCondition returns Completed condition from imageclusterinstall.
+func (builder *ImageClusterInstallBuilder) GetCompletedCondition() (*hivev1.ClusterInstallCondition, error) {
```

Comment:
I think you can simplify  next 4 function by adding getCondition function which accepts condition argument. 
 GetCompletedCondition,GetFailedCondition,GetRequirementsMetCondition,GetStoppedCondition can re-use getCondition with different argument. WDYT ?